### PR TITLE
[ESP32] Fixed lock app crash

### LINKED ISSUE #25874: [BUG] [ESP32C3][lock-app] kernel panic
### Reproduction steps

1. cd ${MATTER_SDK}
2. source scripts/activate.sh
3. cd ${IDF_PATH}
4. source export.sh
5. cd ${MATTER_SDK}/examples/lock-app/esp32
6. idf.py set-target esp32c3
7. idf.py build
8. idf.py erase-flash flash
9. idf.py monitor
 then, kernel panic occurs.
[esp32c3.log](https://github.com/project-chip/connectedhomeip/files/11097014/esp32c3.log)

 

### Bug prevalence

always

### GitHub hash of the SDK that was being used

commit def9802275c65ce678217a55574ba70e1367d1a4 (HEAD -> master, origin/master, origin/HEAD)

### Platform

esp32

### Platform Version(s)

_No response_

## Environment
Host OS : Ubuntu 22.04
ESP-IDF: v4.4.4
Target Board : ESP32-C3-devKitM-1

### PR DESCRIPTION
**Pull request by jadhavrohit924**
Problem:

- Fixes #25874 and #24913.

Changes:

- See above.

Testing:

- Tested commissioning and cluster control.

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/lock-app/esp32/main/AppTask.cpp [hunk 1]
```diff
@@ -86,7 +85,10 @@ CHIP_ERROR AppTask::Init()
                                   (void *) this,    // init timer id = app task obj context
                                   TimerEventHandler // timer callback handler
     );
+
+    chip::DeviceLayer::PlatformMgr().LockChipStack();
```
[the remark this answers is not in the document]
Comment (jadhavrohit924) [reply examples/lock-app/esp32/main/AppTask.cpp]: Dropped prefix

**Review by dhrishi — APPROVED** [review]

File: examples/lock-app/esp32/main/AppTask.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (andy31415) [reply examples/lock-app/esp32/main/AppTask.cpp]: How about RAII?

**Comment by karlp** [comment]
I'm still seeing this on the v1.1-branch, which includes this fix, any other ideas?

**Comment by karlp** [comment]
Ok I've got a fix for this I believe.  My commit https://github.com/project-chip/connectedhomeip/pull/27270 was developed and tested on the matter 1.0 branch, and the code was identical in the master/1.1 branches, and obviously missing, so was jsut committed there.  However, the lock checking must have been added later, which then breaks it again.  I'll make a new PR tomorrow for this.  It's quite disconcerting just how different all the different AppTask.cpp files are given that they are all meant to be ~equivalent.  I suspect many of the other lock examples are probably also crashing on this on other platforms...

**Pull request merged**